### ISSUE TITLE
Script for Firmware Update

### ISSUE THREAD

**Issue by FlorianP2023** [netfishers-onl/Netshot#265]
Hello,
I would like to implement a simple task for firmwareupdate.

The following command does the copy task: "copy tftp://1.1.1.1/firmware.dat backup"
But you have to confirm by pressing "y" afterwards (on the CLI). I dont know how to implement this.
Is there an example for this? My implementation doesn't work since it is not sending "y" to confirm.

function run(cli, device) {
	cli.macro("configure");
	cli.command("copy tftp://10.10.10.10/N1100v6.7.1.21.stk backup");
	cli.macro("end");
} 

In addition the copy task will take up to 5 min. Will I run into a timeout? (and how to avoid this?)
Any help is appreciated.
Regards

**Comment by FlorianP2023** [comment]
Hello again,
I managed to put the firmware update command in a macro.
But this it is static, but I have to use different filenames for different firmware image (and also in future).

So is there a way to interact with the switch cli using the script (in the window)?

For example, I send
function run(cli, device) {
	cli.macro("enable");
	cli.command("copy tftp://10.9.112.75/N1100v6.7.1.21.stk backup");
        cli.command("boot system backup");
}
The switch CLI response is "Are you sure you want to start? (y/n) ". The script stops here and runns into a timeout.
How to send the neccessary "y"? Any idea?
I have similiar problems with other commands that have to be confirmed. I'm sure there is way but I don't see it.
Again, any help is very much appreciated.
See also attached log.

Regards
[fwupdate-log.txt](https://github.com/netfishers-onl/Netshot/files/12092212/fwupdate-log.txt)

**Comment by SCadilhac** [comment]
Macros with params could be useful indeed.
For the time being you should be able to do something like:

```js
cli.command("copy tftp://10.9.112.75/N1100v6.7.1.21.stk backup", {
  mode: {
    prompt: /Are you sure you want to start? \(y\/n\)/,
  },
});
cli.command("y");
```

**Comment by FlorianP2023** [comment]
Thank you very much.

I will try when I'm back in the office.

Regards

**Comment by FlorianP2023** [comment]
Thank you very much, it works now and I could update my switches on the fly (over 200).

Is there a documentation which commands I can use in the scripts?
I added one item to the device called "Last Snapshot" and wanted to fill it with the date of the snapshot (during the snapshot). 

I used for this 
device.set("lastSnapshot", getDate());
but it fails. How can I populate this field with the actual date and time?

Regards,
Florian
